website/docs/04_design/package-config.md

```diff
@@ -0,0 +1,167 @@
+  - **`Label`** (`string`):
+    The label is used to denote an input field related to this value in a UI.
+    By default the name of the value should be used.
+  - **`Description`** (`string`):
+    The description can be used to give more context to a value.
+  - **`Hints`** (`string` enum):
+    Hints offer package maintainers the ability to make some elements of the UI more prominent.
+    For example, every value can be set to reference the value of a secret key, but if a value has the
+    "SuggestSecretRef" hint, this option can be highlighted by the UI or enabled by default.
+    _Available hints and whether they will be included in the initial release is TBD_
+- **`DefaultValue`** (`string`):
+  The default value is pre-selected/pre-filled in the form field of this value for new packages.
+- **`Options`** (`[]string`):
+  Available choices for values of type options.
+  Should be ignored for other types.
+- **`Constraints`**:
+  Specifying a number of constraints is possible.
+  Available constraints are
+  `Required` (`bool`), `Min` (`int`), `Max` (`int`), `MinLength` (`int`), `MaxLength` (`int`), `Pattern` (`string`).
+  These should be checked by the UI, as well as by the validating webhook.
+  Not all constrains apply to all types of value. Non-applicable constraints are ignored.
+  For example a "text" value with constraints.max = 3 is the same as a "text" value with no constraints.
+- **`Targets`**:
+  Where to apply this value.
+  Either the path of a helm value, or a resource identifier (`ApiVersion`, `Kind`, `Name`, `Namespace`) combined with patch information.
+  Initially, the idea is to use RFC 6902 JSON patches but this is still TBD.
+  We use Unstructured for plain resources, which already supports setting values via a kind of JSON path,
+  but it does not support setting values in lists.
+  So, for example, it would not be possible to change something in the container of a deployment, since the containers are a list.
+  Maybe [evanphx/json-patch](https://github.com/evanphx/json-patch) can be a useful alternative, but it only works on byte slices.
+
+The `PackageSpec` has a property `Values` of type `map[string]ValueConfiguration`.
+The key in this map used to identify the corresponding `ValueSpecification` with the same name in the `PackageManifest`s `Values` map.
+A `ValueConfiguration` must have exactly one of the following properties:
+
+- **`Value`**:
+  A literal value.
+- **`ValueFrom`**:
+  To represent a reference value. `ValueFrom` must have exactly one of the following properties:
+  - **`ConfigMapRef`**:
+    To reference a `Key` of a ConfigMap with `Name` in `Namespace`
+  - **`SecretRef`**:
+    To reference a `Key` of a Secret with `Name` in `Namespace`
+  - **`PackageRef`**:
+    To reference the value of the `ValueConfiguration` with name `Value` of a package with `Name`.
+
+## Examples
+
+```yaml title="PackageManifest with a simple value specification"
+name: foo
+values:
```

Comment:
you already mentioned that you haven't been sure about calling the `Package` and `PackageManifest` attribute values. I would suggest renaming the `PackageManifest` value definition to either:

 - valueSpec
 - valueManifest
 - valueDefinition
 - .. or something else

Reply:
As discussed, it will be changed to `valueDefinitions`